packages/relay/src/RelayServer.ts

```diff
@@ -688,12 +692,32 @@ latestBlock timestamp   | ${latestBlock.timestamp}
     await this.withdrawToOwnerIfNeeded(block.number, block.hash, currentBlockTimestamp)
     this.lastRefreshBlock = block.number
     await this._refreshGasFees()
-    await this.registrationManager.refreshBalance()
-    if (!this.registrationManager.balanceRequired.isSatisfied) {
-      this.setReadyState(false)
+    const managerBalanceNotReady = await this._refreshAndCheckBalances()
+    if (managerBalanceNotReady) {
       return []
     }
-    return await this._handleChanges(block, currentBlockTimestamp)
+    return await this._handleChanges(block)
+  }
+
+  async _refreshAndCheckBalances (): Promise<boolean> {
+    const minBalanceToNotReadyFactor = 2
+    let managerBalanceNotReady = false
+    let workerBalanceNotReady = false
+    if (this.shouldRefreshBalances) {
+      await this.registrationManager.refreshBalance()
+      this.workerBalanceRequired.currentValue = await this.getWorkerBalance(0)
+
+      managerBalanceNotReady = this.registrationManager.balanceRequired.currentValue.lt(toBN(this.config.managerMinBalance.toString()).divn(minBalanceToNotReadyFactor))
+      workerBalanceNotReady = this.workerBalanceRequired.currentValue.lt(toBN(this.config.workerMinBalance.toString()).divn(minBalanceToNotReadyFactor))
+      if (managerBalanceNotReady || workerBalanceNotReady) {
+        this.logger.debug(`${managerBalanceNotReady ? 'manager' : 'worker'} balance too low`)
+        this.setReadyState(false)
+      }
+      const managerBalanceAwaitsReplenish = this.registrationManager.balanceRequired.currentValue.lt(toBN(this.config.managerMinBalance.toString()))
```

Comment:
"awaits" assumes there is a pending transaction and we are 'waiting' for it to resolve.

Please rename to 'requires'.